indico/web/client/js/react/components/PrincipalSearchField.jsx

```diff
@@ -94,12 +120,18 @@ export default class UserSearchField extends React.Component {
         this.setState({
             isFetching: true
         });
-        const query = (searchQuery.indexOf('@') !== -1)
-            ? {email: searchQuery}
-            : {name: searchQuery};
-        const users = await searchUser(query);
-        const options = users.map(this.renderUser);
-        this.userCache = Object.assign(this.userCache, ...users.map(user => ({[user.id]: user})));
+
+        const query = (searchQuery.indexOf('@') !== -1) ? {email: searchQuery} : {name: searchQuery};
+        const {withGroups} = this.props;
+        let items = [];
+
+        items = items.concat(await searchUser(query));
+        if (withGroups) {
```

Comment:
You are running the two searches sequentially. What about using `Promise.all(...)` to save time?